packages/next-codemod/bin/upgrade.ts

```diff
@@ -600,3 +617,119 @@ function writeOverridesField(
     }
   }
 }
+
+function getDependenciesOutOfRange(
+  appPackageJson: any,
+  versionMapping: Record<string, { version: string; required: boolean }>,
+  allDependenciesToInstall: Record<string, string>
+) {
+  const versionMappingKeys = Object.keys(versionMapping)
+  const dependenciesOutOfRange = new Map<
+    string,
+    { current: string; expected: string; from: string }
```

Comment:
Makes 100% sense

Reply:
Used `currentVersion` and `expectedVersionRange`!

No `peerPackageName` because it had to become the "key" value of the object. Previously, it deduped the target package name. If peer dep unmet range for "react" in package A and B, only B was listed, so use package name as the key and let them include deps, e.g. "react".

```js
{
  'package-a': {
    'react': {
      currentVersion: '18.2.0',
      expectedVersionRange: '^17.0.0'
    },
    'react-dom': {
      currentVersion: '18.2.0',
      expectedVersionRange: '^17.0.0'
    }
  },
  'package-b': {
    'react': {
      currentVersion: '13.0.0',
      expectedVersionRange: '^12.0.0'
    }
  }
}
```